pyccel/codegen/printing/pycode.py

```diff
@@ -124,13 +124,20 @@ def _print_FunctionDef(self, expr):
         return code
 
     def _print_Return(self, expr):
-        code = ''
+        rhs_list = []
+        lhs_list = []
+        expr_return_vars = []
         if expr.stmt:
-            code += self._print(expr.stmt)+'\n'
-        if expr.expr:
-            ret = ','.join([self._print(i) for i in expr.expr])
-            code += 'return {}'.format(ret)
-        return code
+            for i in expr.stmt.body:
+                rhs_list.append(i.rhs)
+                lhs_list.append(i.lhs)
+        for a in expr.expr:
+            if lhs_list and a not in lhs_list:
+                expr_return_vars.append(a)
+            elif not lhs_list:
+                expr_return_vars.append(a)
```

Comment:
Can this be reduced to a list comprehension (list comprehensions run a lot faster than for loops):
```
expr_return_vars = [a if not lhs_list or a not in lhs_list for a in expr.expr]
```
?